### ISSUE TITLE
The effect causes KWin to paint the area behind opaque windows, resulting in higher GPU usage

### ISSUE THREAD

**Issue by taj-ny** [matinlotfali/KDE-Rounded-Corners#207]
With just a few blurred windows in the background, the regions that have to be painted quickly add up, even if all the windows are covered by an opaque window.

I believe this is an accident, as ``data.setTranslucent()`` sets ``data.opaque`` to an empty region, but just a few lines later you are subtracting from that empty region.
https://github.com/matinlotfali/KDE-Rounded-Corners/blob/master/src/ShapeCornersEffect.cpp#L137-L149
https://github.com/KDE/kwin/blob/v6.0.3/src/effect/effect.cpp#L20-L25

I commented out the ``data.setTranslucent()`` line and the GPU usage has dropped in the situation described at the beginning. I haven't noticed any issues with corners, shadows or the outline on Wayland and X11.

Although I should mention that Plasma 6 broke transparent color schemes and the translucent area behind some windows is no longer painted ([bug report](https://bugs.kde.org/show_bug.cgi?id=480608), [video](https://www.reddit.com/r/kde/comments/1bbdkus/what_the_hell_is_this_happens_on_every)), resulting in artifacts on Wayland. ``data.setTranslucent()`` does fix this, so removing it could bring back the issue for some users.

I'm running Plasma 6.0.3 with KDE-Rounded-Corners 0.6.1.

**Comment by matinlotfali** [comment]
You are absolutely correct. The `setTranslucent` is there only because of the ugly artifacts. I should have written comments in the code about it.

**Comment by matinlotfali** [comment]
Sadly, I don't know how else I can prevent the artifact. I am open for suggestions.

**Comment by rkunschke** [comment]
I have two questions about that: 

Is that problem only with semi-transparent and transparent windows?

And would it be a good idea to provide a option to disable that behavier?
Maybe with a warning about artifacts. Maybe on some setups the artifacts seem to be no problem as @taj-ny descried: "I haven't noticed any issues with corners, shadows or the outline on Wayland and X11."

**Comment by matinlotfali** [comment]
Sure I can make it an option defaulting to false.

**Comment by matinlotfali** [comment]
@rkunschke Can you test #434?

**Comment by taj-ny** [comment]
> Is that problem only with semi-transparent and transparent windows?

Only windows with opaque areas are affected.

> I commented out the data.setTranslucent() line and the GPU usage has dropped in the situation described at the beginning. I haven't noticed any issues with corners, shadows or the outline on Wayland and X11.

After a year of using this I did notice some artifacts in the corners that appear rarely, so that's not the correct solution. I don't have time for testing right now but perhaps try this instead of ``data.setTranslucent()``:
```cpp
data.mask |= Effect::PAINT_WINDOW_TRANSLUCENT;
data.mask &= ~Effect::PAINT_WINDOW_OPAQUE;
```

This marks the window as translucent but doesn't reset the opaque region (https://github.com/KDE/kwin/blob/580a077e94f35c73417de9e880508f1c0b5d8dbd/src/effect/effect.cpp#L20).

**Comment by matinlotfali** [comment]
@rkunschke Can you also test https://github.com/matinlotfali/KDE-Rounded-Corners/pull/435?

**Comment by rkunschke** [comment]
> [@rkunschke](https://github.com/rkunschke) Can you also test [#435](https://github.com/matinlotfali/KDE-Rounded-Corners/pull/435)?

I've tested it and felt no difference. Is there a indication if it worked?

**Comment by matinlotfali** [comment]
Did it reduce the GPU usage?

I will also test it on KWin 5.27 to see if there is any artifact showing up on either PRs

**Comment by rkunschke** [comment]
> Did it reduce the GPU usage?
> 
> I will also test it on KWin 5.27 to see if there is any artifact showing up on either PRs

No. But I am not sure how to test this correnctly.
I've tested it with 11 konsole windows on top of each other. GPU utilization was at 3-6 percent in both cases (Skylake igu notebook).

Is the effect only measurable when windows are transparent or something? Does I need to measure when animations are running?

**Comment by rkunschke** [comment]
I've tried to measure the effect when stacking lots of windows over each other. It also made no difference. But I am also getting no artifacts.
When I stack multiple opaque windows, the windows which are covered shouldn't be rendered I guess? That works for me when they're minimized or on another workspace. 
But I still think I am measuring the wrong thing.

But for my understanding the change can be still released. It's optional an not making any problems for me and @taj-ny .
